src/pages-and-resources/discussions/data/api.js

```diff
@@ -118,6 +121,12 @@ function denormalizeData(courseId, appId, data) {
   if (data.launchUrl) {
     ltiConfiguration.lti_1p1_launch_url = data.launchUrl;
   }
+  if ('piiShareUsername' in data) {
```

Comment:
If data is serialized from the form, how these keys will be excluded during serialization? I am just trying to understand why you need to check key is `in` data?

Reply:
The reason for that is that we don't support setting these values if a course doesn't have PII sharing enabled. Doing so will raise an error even if other values are OK. 

If pii sharing isn't enabled, the form will not show up, so these values will not be in the data, and as such won't be sent to the server.